### PR TITLE
Sanitize log messages

### PR DESCRIPTION
**Pull request by rsajwani**
Signed-off-by: Rameez Sajwani <rameezwazirali@hotmail.com>

## Description

This is an effort to sanitize and redact PII from error logging.

Important Note: I am not able to use *terseErrors (vtgate-config-terse-errors) , given its defined in vtgate package and using it in shared_buffer.go results in circular dependency. It will require some refactoring which I believe we can do in separate task.

## Checklist

-   [ ] "Backport me!" label has been added if this change should be backported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by vmg — APPROVED** [review]
Looks good! Thanks for the backport!

**Review by deepthi — COMMENTED** [review]
Can you change the file names to follow the convention? underscores (aka snake case), not camelCase.
For the rest, @systay can you review as well before we merge?

File: go/vt/vtgate/errorsanitizer/errorSanitizer.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	// Truncate errors longer than this
+	maxErrorLength = 256
+)
+
+var (
+	// Remove any BindVars or Sql, and anything after it
+	reTruncateError = regexp.MustCompile(`[,:] BindVars:|[,:] Sql:`)
+
+	// Keep code = foo, Duplicate entry, or syntax error, but remove anything after it
+	reTruncateErrorAfter = regexp.MustCompile(`code = \S+|Duplicate entry|syntax error at position \d+`)
```
Comment (deepthi) [inline go/vt/vtgate/errorsanitizer/errorSanitizer.go:33]: If there are other error messages that don't match these patterns, we won't be truncating, right?

Comment (systay) [reply go/vt/vtgate/errorsanitizer/errorSanitizer.go:33]: that is my understanding

**Comment by deepthi** [comment]
@rsajwani since the PR was merged without addressing my feedback, can you please create a follow up PR?

**Comment by rsajwani** [comment]
> 
for renaming 

https://github.com/vitessio/vitess/pull/10406

**Pull request merged**